src/defrag/defrag.cpp

```diff
@@ -788,7 +815,7 @@ void Defrag::poll()
 	dprintf(D_ALWAYS,"Looking for %d machines to drain.\n",num_to_drain);
 
 	ClassAdList startdAds;
-	std::string requirements = "PartitionableSlot && Offline =!= true && Draining =!= true";
+	std::string requirements(BASE_SLOT_CONSTRAINT " && Draining =!= true");
```

Comment:
Would it make sense for `Draining =!= true` to be a constant defined near `BASE_SLOT_CONSTRAINT` and/or `BASE_DRAINING_CONSTRAINT`?  Seems like you might want to make sure the machines-to-drain query and the machines-not-to-drain query are consistent (or, ideally, constructed from each other).